easybuild/tools/containers/docker.py

```diff
@@ -0,0 +1,179 @@
+
+RUN mkdir /app && \\
+    mkdir /scratch && \\
+    mkdir /scratch/tmp && \\
+    useradd -m -s /bin/bash easybuild && \\
+    chown easybuild:easybuild -R /app && \\
+    chown easybuild:easybuild -R /scratch
+
+USER easybuild
+
+RUN set -x && \\
+    . /usr/share/lmod/lmod/init/sh && \\
+    eb %(eb_opts)s --installpath=/app/ --prefix=/scratch --tmpdir=/scratch/tmp
+
+RUN touch ${HOME}/.profile && \\
+    echo '\\n# Added by easybuild docker packaging' >> ${HOME}/.profile && \\
+    echo 'source /usr/share/lmod/lmod/init/bash' >> ${HOME}/.profile && \\
+    echo 'module use %(init_modulepath)s' >> ${HOME}/.profile && \\
+    echo 'module load %(mod_names)s' >> ${HOME}/.profile
+
+CMD ["/bin/bash", "-l"]
+"""
+
+_DOCKER_TMPLS = {
+    DOCKER_BASE_IMAGE_UBUNTU: DOCKER_UBUNTU1604_TMPL,
+    DOCKER_BASE_IMAGE_CENTOS: DOCKER_CENTOS7_TMPL,
+}
+
+
+class DockerContainer(ContainerGenerator):
+
+    TOOLS = {'docker': None, 'sudo': None}
+
+    RECIPE_FILE_NAME = 'Dockerfile'
+
+    def __init__(self, *args, **kwargs):
+        super(DockerContainer, self).__init__(*args, **kwargs)
+        # NOTE (med): set default value for _container_base
+        self._container_base = self._container_base or DEFAULT_DOCKER_BASE_IMAGE
+
+    def resolve_template(self):
+        return _DOCKER_TMPLS[self._container_base]
+
+    def resolve_template_data(self):
+        os_deps = det_os_deps(self._easyconfigs)
+
+        module_naming_scheme = ActiveMNS()
+
+        ec = self._easyconfigs[-1]['ec']
+
+        init_modulepath = os.path.join("/app/modules/all", *module_naming_scheme.det_init_modulepaths(ec))
+
+        mod_names = [e['ec'].full_mod_name for e in self._easyconfigs]
+
+        eb_opts = [os.path.basename(ec['spec']) for ec in self._easyconfigs]
+
+        return {
+            'os_deps': ' '.join(os_deps),
+            'eb_opts': ' '.join(eb_opts),
+            'init_modulepath': init_modulepath,
+            'mod_names': ' '.join(mod_names),
+        }
+
+    def validate(self):
+        super(DockerContainer, self).validate()
+        if self._container_base not in [DOCKER_BASE_IMAGE_UBUNTU, DOCKER_BASE_IMAGE_CENTOS]:
+            raise EasyBuildError("Unsupported container base image '%s'" % self._container_base)
+
+    def build_image(self, dockerfile):
+        ec = self._easyconfigs[-1]['ec']
+
+        module_naming_scheme = ActiveMNS()
+        module_name = module_naming_scheme.det_full_module_name(ec)
+
+        tempdir = tempfile.mkdtemp(prefix='easybuild-docker')
+        container_name = build_option('container_image_name') or "%s:latest" % module_name.replace('/', '-')
+        docker_cmd = ' '.join(['sudo', 'docker', 'build', '-f', dockerfile, '-t', container_name, '.'])
+
+        print_msg("Running '%s', you may need to enter your 'sudo' password..." % docker_cmd)
+        run_cmd(docker_cmd, path=tempdir, stream_output=True)
+        print_msg("Docker image created at %s" % container_name, log=_log)
+
+        shutil.rmtree(tempdir)
```

Comment:
please use `rmtree2` from `easybuild.tools.filetools` (the function name is silly, but it does more than just plain `rmtree`)